Editor/CoreAPI/GameLiftComputeManager.cs

```diff
@@ -19,7 +22,37 @@ public GameLiftComputeManager(IAmazonGameLiftWrapper wrapper)
             _amazonGameLiftWrapper = wrapper;
         }
 
-        public async Task<RegisterFleetComputeResponse> RegisterFleetCompute(string computeName, string fleetId, string fleetLocation,
+        public async Task<RegisterFleetComputeResponse> RegisterFleetCompute(string computeName, string fleetId,
+            string fleetLocation,
+            string ipAddress)
+        {
+            if (_amazonGameLiftWrapper != null)
+            {
+                if (string.IsNullOrWhiteSpace(computeName))
+                {
+                    return Response.Fail(new RegisterFleetComputeResponse
+                    {
+                        ErrorCode = ErrorCode.InvalidComputeName
+                    });
+                }
+                
+                if (string.IsNullOrWhiteSpace(ipAddress))
+                {
+                    return Response.Fail(new RegisterFleetComputeResponse
+                    {
+                        ErrorCode = ErrorCode.InvalidIpAddress
+                    });
+                }
+                
+                var registerFleetComputeResponse = await RegisterCompute(computeName, fleetId, fleetLocation, ipAddress);
+
+                return registerFleetComputeResponse;
+            }
+
+            return Response.Fail(new RegisterFleetComputeResponse { ErrorCode = ErrorCode.AccountProfileMissing });
```

Comment:
Let's keep the error handling within the existing `RegisterCompute` method, there doesn't seem to be a need to add a separate method here.  Also the Failure code seems arbitrary. The failure handling in lines 79-83 looks right from an initial glance and should be retained.

Reply:
Merged everything into `RegisterFleetCompute` along with the if-guard pattern to keep it clean.